arangod/Cluster/EnsureIndex.cpp

```diff
@@ -150,9 +155,11 @@ bool EnsureIndex::first() {
       // continue with the job normally
     }
 
+    auto lambda = std::make_shared<std::function<arangodb::Result(uint64_t u)>>(
+        [this](uint64_t u) { return setProgress(u); });
```

Comment:
Why couldn't you pass just function?

Reply:
Only a matter of preference. There are no performance implications. Will keep it as is, if you allow.